New glossary: logical and physical properties (#33860)
mdn/content: files/en-us/glossary/logical_properties/index.md

---
title: Logical properties
slug: Glossary/Logical_properties
page-type: glossary-definition
---

{{GlossarySidebar}}

> CSS **logical properties** provide a way to style layouts based on the document's writing mode and direction rather than the physical dimensions of the viewport. This enables more flexible and maintainable designs, especially for websites supporting multiple languages.

Reviewer: ```suggestion
CSS **logical properties** provide a way to lay out content based on the document's writing mode and direction rather than the physical dimensions of the viewport. This enables more flexible and maintainable designs, especially for websites supporting multiple languages.
```

> Properties like {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("bottom")}}, and {{cssxref("left")}} are **physical properties**. They define positions based on physical directions; with top, right, bottom, and left referring to the physical dimensions of the viewport. Physical properties style features by physical direction, such as {{cssxref("margin-top")}}, {{cssxref("border-right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}. These are **physical** as they reference specific sides of an element.

Reviewer: It feels like you are saying the same thing twice in the two sentences of this paragraph, unless I'm missing something. Can you try to clarify this, or amalgamate the two together?

Logical properties are relative to the content flow and use directional keywords that are relative to the block and inline axes.

> The **block axis** refers to the axis that defines the stacking order of elements in a block layout. It's essentially the direction along which blocks of content, like paragraphs ({{htmlelement("p")}}), headings, and divs ({{htmlelement("div")}}), are laid out on a webpage. This is the vertical direction of the content flow, going from top to bottom, in left-to-right and left-to-right languages. The **block-start** and **block-end** directions is the "from" and "to" directions, respectively, with `block-start` being the equivalent of `top` and `block-end` being the equivalent of `bottom`. In Japanese

Reviewer: ```suggestion
The **block axis** refers to the axis that defines the stacking order of elements in a block layout. It's essentially the direction along which blocks of content, like paragraphs ({{htmlelement("p")}}), headings, and divs ({{htmlelement("div")}}), are laid out on a webpage. This is the vertical direction of the content flow, going from top to bottom, in left-to-right and left-to-right languages. **block-start** and **block-end** represent the start edge and end edge of content along the block axis, respectively, with `block-start` being the equivalent of `top` and `block-end` being the equivalent of `bottom`. In Japanese
```

Reviewer: You've also got a last sentence "In Japanese..." that needs completing.

> The **inline axis** is perpendicular to the block axis. The inline axis represents the direction along which inline content like text flows within a block. This is the horizontal direction of the content flow, going from left to right in left-to-right writing modes, like English, and right to left in right-to-left languages, like Arabic and Hebrew. The **inline-start** and **inline-end** direction depend on the writing direction, with the values and properties `inline-start` and `inline-end` being the equivalent of `left` and `right` properties and values, respectively, for left-to-right languages, and `right` and `left` in right-to-left writing modes.

Reviewer: ```suggestion
The **inline axis** is perpendicular to the block axis. The inline axis represents the direction along which inline content like text flows within a block. This is the horizontal direction of the content flow, going from left to right in left-to-right writing modes, like English, and right to left in right-to-left languages, like Arabic and Hebrew. The **inline-start** and **inline-end** directions depend on the writing direction. `inline-start` and `inline-end` represent the start edge and end edge of content along the inline axis, respectively. They are the equivalent of `left` and `right` for left-to-right writing modes, and `right` and `left` in right-to-left writing modes.
```

---
title: Logical properties
slug: Glossary/Logical_properties
page-type: glossary-definition
---

{{GlossarySidebar}}

CSS **logical properties** provide a way to lay out content based on the document's writing mode and direction rather than the physical dimensions of the viewport. This enables more flexible and maintainable designs, especially for websites supporting multiple languages.

> Whereas {{glossary("physical properties")}}, like {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}, define positions and features based on physical directions and reference specific sides of an element, logical properties such as {{cssxref("block-start")}}, {{cssxref("inline-end")}}, {{cssxref("padding-block-end")}}, and {{cssxref("border-end-end-radius")}} use logical directional keywords that are relative to the content flow along the block and inline axes.

Author: ```suggestion
Whereas {{glossary("physical properties")}}, like {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}, define positions and features based on physical directions and reference specific sides of an element, logical properties such as {{cssxref("inset-block-start")}}, {{cssxref("inset-inline-end")}}, {{cssxref("padding-block-end")}}, and {{cssxref("border-end-end-radius")}} use logical directional keywords that are relative to the content flow along the block and inline axes.
```

---
title: Logical properties
slug: Glossary/Logical_properties
page-type: glossary-definition
---

{{GlossarySidebar}}

CSS **logical properties** provide a way to lay out content based on the document's writing mode and direction rather than the physical dimensions of the viewport. This enables more flexible and maintainable designs, especially for websites supporting multiple languages.

> Whereas {{glossary("physical properties")}} define positions based on physical directions; with {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("bottom")}}, and {{cssxref("left")}} referring to the physical dimensions of the viewport and properties like {{cssxref("margin-top")}}, {{cssxref("border-right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}reference specific sides of an element, logical properties are relative to the content flow and use directional keywords that are relative to the block and inline axes.

Reviewer: ```suggestion
Whereas {{glossary("physical properties")}} define positions based on physical directions, logical properties such as {{cssxref("margin-block-start")}} and {{cssxref("padding-inline-end")}} use logical directional keywords relative to the content flow along the block and inline axes.
```

I don't think all of the physical examples are needed. You link to the physical properties page, where you provide examples of such properties. The original was rather long and cumbersome to read.

I have also included the information provided by the sentence above the "See also" section, which I am suggesting you delete.

Author: Went with:

```
Whereas {{glossary("physical properties")}}, like {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}, define positions and features based on physical directions and reference specific sides of an element, logical properties such as {{cssxref("block-start")}}, {{cssxref("inline-end")}}, {{cssxref("padding-block-end")}}, and {{cssxref("border-end-end-radius")}} use logical directional keywords that are relative to the content flow along the block and inline axes.
```

using 4 physical properties and their mapped logical property equivalents. I think that improves it a lot.

Reviewer: Much better — my only issue is that I don't think `block-start` and `inline-end` are properties. Did you mean `margin-block-start` and `margin-inline-end`, or similar?

Reviewer: The links are broken, you see. Of course, we might just not have documented them yet...

> Whereas {{glossary("physical properties")}} define positions based on physical directions; with {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("bottom")}}, and {{cssxref("left")}} referring to the physical dimensions of the viewport and properties like {{cssxref("margin-top")}}, {{cssxref("border-right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}}reference specific sides of an element, logical properties are relative to the content flow and use directional keywords that are relative to the block and inline axes.

Author: ```suggestion
Whereas {{glossary("physical properties")}}, like {{cssxref("top")}}, {{cssxref("right")}}, {{cssxref("padding-bottom")}}, and {{cssxref("border-bottom-left-radius")}},  efine positions based on physical directions and reference specific sides of an element, logical properties logical properties such as {{cssxref("margin-block-start")}} and {{cssxref("padding-inline-end")}} use logical directional keywords that are relative to the content flow along the block and inline axes.
```

## Block direction

> The **block axis** refers to the axis that defines the stacking order of elements in a block layout. It's essentially the direction along which blocks of content, or **block direction**, like paragraphs ({{htmlelement("p")}}), headings, and divs ({{htmlelement("div")}}), are laid out on a webpage. This is the vertical direction of the content flow, going from top to bottom, in left-to-right and left-to-right languages.

Reviewer: ```suggestion
The **block axis** refers to the axis that defines the stacking order of elements in a block layout. It's essentially the direction along which blocks of content — like paragraphs ({{htmlelement("p")}}), headings, and divs ({{htmlelement("div")}}) — are laid out on a webpage. This is also known as the **block direction**. In left-to-right and left-to-right languages, the block direction is the vertical direction of the content flow, going from top to bottom.
```

The **block-start** and **block-end** directions represent the _start edge_ and _end edge_ of content along the block axis, or the "from" and "to" directions, respectively, with `block-start` being the equivalent of `top` and `block-end` being the equivalent of `bottom` in horizontal writing modes.

## Inline direction

> The **inline axis** is perpendicular to the block axis. The inline axis represents the direction along which inline content, or **inline direction**, like text flows within a block. This is the horizontal direction of the content flow, going from left to right in left-to-right writing modes, like English, and right to left in right-to-left languages, like Arabic and Hebrew.

Reviewer: ```suggestion
The **inline axis** is perpendicular to the block axis. The inline axis represents the direction along which inline content such as text flows within a block. This is also known as the **inline direction**. In left-to-right writing modes, like English, the inline direction is horizontal, left-to-right. In right-to-left languages, like Arabic and Hebrew, the inline direction is horizontal, right-to-left.
```

> **Inline-start** and **inline-end** represent the _start edge_ and _end edge_ of content along the inline axis, respectively, with the values and properties `inline-start` and `inline-end` being the equivalent of `left` and `right` properties and values in horizontal writing modes, but whether each means `right` or `left` depends on the writing direction (`inline-start` is `left` in left-to-right languages and `right` in right-to-left languages.)

Reviewer: ```suggestion
**Inline-start** and **inline-end** represent the _start edge_ and _end edge_ of content along the inline axis, respectively, with the values and properties `inline-start` and `inline-end` being equivalent to `left` and `right` properties and values in horizontal writing modes. Which ones are equivalent to `right` or `left` depends on the writing direction — for example `inline-start` is `left` in left-to-right languages and `right` in right-to-left languages.
```

> Therefore, logical properties, like {{cssxref("margin-block-start")}} and {{cssxref("padding-inline-end")}}, use logical keywords relative to the content flow.

Reviewer: I'd suggest removing this and moving these details as per my suggestion above. This sentence feels a bit out of place and without context, in the current position.

Author: done

> 
> Therefore, logical properties, like {{cssxref("margin-block-start")}} and {{cssxref("padding-inline-end")}}, use logical keywords relative to the content flow.

Author: ```suggestion
```
